sstables/mx/writer.cc

```diff
@@ -991,13 +991,23 @@ void writer::maybe_record_large_rows(const sstables::sstable& sst, const sstable
 }
 
 void writer::maybe_record_large_cells(const sstables::sstable& sst, const sstables::key& partition_key,
-        const clustering_key_prefix* clustering_key, const column_definition& cdef, uint64_t cell_size) {
-    auto& entry = _large_data_stats.map.at(large_data_type::cell_size);
-    if (entry.max_value < cell_size) {
-        entry.max_value = cell_size;
+        const clustering_key_prefix* clustering_key, const column_definition& cdef, uint64_t cell_size, uint64_t collection_items) {
+    auto& cell_size_entry = _large_data_stats.map.at(large_data_type::cell_size);
+    if (cell_size_entry.max_value < cell_size) {
+        cell_size_entry.max_value = cell_size;
     }
+    auto& collection_items_entry = _large_data_stats.map.at(large_data_type::items_in_collection);
+    if (collection_items_entry.max_value < collection_items) {
+        collection_items_entry.max_value = collection_items;
+    }
+    // FIXME: pass collection_items to large_data_handler
     if (_sst.get_large_data_handler().maybe_record_large_cells(_sst, *_partition_key, clustering_key, cdef, cell_size).get0()) {
-        entry.above_threshold++;
+        if (cell_size > cell_size_entry.threshold) {
+            cell_size_entry.above_threshold++;
+        }
+        if (collection_items > collection_items_entry.threshold) {
+            collection_items_entry.above_threshold++;
+        }
     };
```

Comment:
Note: this looks very inefficient, looking up something in a hash only to not do anything with it.

Reply:
We conditionally update the entries' `max_value` after the lookup. That hasn't changed.
Maybe it should have been a disk array to save the map lookup, but then it would be harder to extend as a disk format.

Reply:
Sure, it was inefficient before that patch too.

We should use separate member variables, and convert to disk_hash only when writing at the end.

Reply:
ok, will do as follow up

Reply:
https://github.com/scylladb/scylladb/issues/11686